jmxtrans-output/jmxtrans-output-elastic/src/main/java/com/googlecode/jmxtrans/model/output/elastic/ElasticWriter.java

```diff
@@ -0,0 +1,237 @@
+			generator.close();
+			return out.toString("UTF-8");
+		} catch (Throwable t) {
+			throw closer.rethrow(t);
+		} finally {
+			closer.close();
+		}
+	}
+
+	@VisibleForTesting
+	void setJestClient(JestClient jestClient) {
+		this.jestClient = jestClient;
+	}
+
+	@Override
+	public void validateSetup(Server server, Query query) throws ValidationException {
+		// no validations
+	}
+
+	@Override
+	public void start() throws LifecycleException {
+		super.start();
+
+		if (jestClient == null) {
+			log.info("Create a jest elastic search client for connection url [{}]", connectionUrl);
+			JestClientFactory factory = new JestClientFactory();
+			factory.setHttpClientConfig(
+					new HttpClientConfig.Builder(connectionUrl)
+							.multiThreaded(true)
+							.build());
+			jestClient = factory.getObject();
+
+		}
+		else {
+			log.info("Note: using injected jestClient instead of creating a new one: [{}]", jestClient);
+		}
+
+		createMapping(jestClient, indexName, TYPE_NAME);
+
+	}
+
+	private static void createMapping(JestClient jestClient, String indexName, String typeName) {
+		synchronized (CREATE_MAPPING_LOCK) {
+			try {
+				IndicesExists indicesExists = new IndicesExists.Builder(indexName).build();
+				boolean indexExists = jestClient.execute(indicesExists).isSucceeded();
+
+				if (!indexExists) {
+
+					CreateIndex createIndex = new CreateIndex.Builder(indexName).build();
+					jestClient.execute(createIndex);
+
+					PutMapping putMapping = new PutMapping.Builder(indexName, typeName,
+							"{\n" +
+									"  \"jmx-entry\": {\n" +
+									"    \"properties\": {\n" +
+									"      \"attributeName\": {\n" +
+									"        \"type\": \"string\",\n" +
+									"        \"index\": \"not_analyzed\"\n" +
+									"      },\n" +
+									"      \"key\": {\n" +
+									"        \"type\": \"string\",\n" +
+									"        \"index\": \"not_analyzed\"\n" +
+									"      },\n" +
+									"      \"metric\": {\n" +
+									"        \"type\": \"string\",\n" +
+									"        \"index\": \"not_analyzed\"\n" +
+									"      },\n" +
+									"      \"server\": {\n" +
+									"        \"type\": \"string\",\n" +
+									"        \"index\": \"not_analyzed\"\n" +
+									"      },\n" +
+									"      \"timestamp\": {\n" +
+									"        \"type\": \"date\"\n" +
+									"      },\n" +
+									"      \"resultAlias\": {\n" +
+									"        \"type\": \"string\",\n" +
+									"        \"index\": \"not_analyzed\"\n" +
+									"      },\n" +
+									"      \"value\": {\n" +
+									"        \"type\": \"float\"\n" +
+									"      }\n" +
+									"    }\n" +
+									"  }\n" +
+									"}").build();
+
+					JestResult result = jestClient.execute(putMapping);
+					if (!result.isSucceeded()) {
+						log.warn("Failed to create mapping: {}", result.getErrorMessage());
+					}
+					else {
+						log.info("Created mapping for index {}", indexName);
+					}
+				}
+			} catch (IOException e) {
```

Comment:
I'm not an expert on ElasticSerach and Jest, but my guess is that if the creation of mappings fail, this output writer will be unuseable. So you should probably just let the exception bubble up and not hide the issue.